android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -361,6 +364,48 @@ constructor(
         ?.find { parentResourceId.equals(it.parentResourceId, ignoreCase = true) }
         ?.count
         ?: 0
+
+    /**
+     * This function sorts [resources] by comparing the values extracted by FHIRPath using the
+     * [fhirPathExpression]. The [dataType] is required for ordering of the. You can optionally
+     * specify the [Order] of sorting.
+     */
+    @JvmOverloads
```

Comment:
very interesting! I hadn't heard of this annotation before, https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-overloads/